extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByPostShuffleFrameProcessor.java

```diff
@@ -233,13 +242,19 @@ private boolean writeOutputRow() throws IOException
     finalizeFn.accept(outputRow);
 
     if (frameWriter.addSelection()) {
+      if (partitionBoostVirtualColumn != null) {
+        partitionBoostVirtualColumn.setValue(partitionBoostVirtualColumn.getValue() + 1);
```

Comment:
Will this cause the compare function on L161 to behave weirdly ?

Reply:
The compare function won't be behaving weirdly because:
1. The boosted column is only a part of the final written frame. The `outpuRow` doesn't contain the boosted column
2. Also, the compare function is computed based on the group by query, which will compare the dims and the __time column (if needed). Since __boost doesn't come there, it will choose to ignore the column altogether.

This can be confirmed in the COUNT(*) tests `testInsertOnExternalDataSource` which aggregates the rows with the correct result.